File: src/gui/IGUIHandler.c
#include "gui/IGUIHandler.r"
#include <assert.h>

#include "gui/IGUIHandler.h"

void event ( void *_self, void *ev ) {
	struct IGUIHandler **self = _self;

	assert ( self );
	if ( (*self)->event ) {
		(*self)->event ( _self, ev );
	}
}

File: include/gui/IGUIHandler.r
#ifndef _IGUIHANDLER_R_
#define _IGUIHANDLER_R_

#include "gui/IGUIObj.r"

#define CONT_IGUIHANDLER \
	CONT_IGUIOBJ \
	void (*event) ( void *view, void *ev );

struct IGUIHandler {
	CONT_IGUIHANDLER
};

#endif
